feature/interface/aggregate/otg_tests/aggregate_forwarding_viable_test/aggregate_forwarding_viable_test.go

```diff
@@ -0,0 +1,596 @@
+	for _, port := range tc.dutPorts {
+		path := gnmi.OC().Interface(port.Name())
+		gnmi.Await(t, tc.dut, path.OperStatus().State(), time.Minute, oc.Interface_OperStatus_UP)
+	}
+}
+
+// configureDUT configures source and destination ports of DUT and creates port-channel as well.
+func (tc *testArgs) configureDUT(t *testing.T) {
+	t.Helper()
+	t.Logf("dut ports = %v", tc.dutPorts)
+	if len(tc.dutPorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got %d", len(tc.dutPorts))
+	}
+
+	d := gnmi.OC()
+
+	if *deviations.AggregateAtomicUpdate {
+		tc.clearAggregate(t)
+		tc.setupAggregateAtomically(t)
+	}
+
+	lacp := &oc.Lacp_Interface{Name: ygot.String(tc.aggID)}
+	if tc.lagType == lagTypeLACP {
+		lacp.LacpMode = oc.Lacp_LacpActivityType_ACTIVE
+	} else {
+		lacp.LacpMode = oc.Lacp_LacpActivityType_UNSET
+	}
+	lacpPath := d.Lacp().Interface(tc.aggID)
+	fptest.LogQuery(t, "LACP", lacpPath.Config(), lacp)
+	gnmi.Replace(t, tc.dut, lacpPath.Config(), lacp)
+
+	agg := &oc.Interface{Name: ygot.String(tc.aggID)}
+	tc.configDstAggregateDUT(agg, &dutDst)
+	aggPath := d.Interface(tc.aggID)
+	fptest.LogQuery(t, tc.aggID, aggPath.Config(), agg)
+	gnmi.Replace(t, tc.dut, aggPath.Config(), agg)
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		fptest.AssignToNetworkInstance(t, tc.dut, tc.aggID, *deviations.DefaultNetworkInstance, 0)
+	}
+
+	srcp := tc.dutPorts[0]
+	srci := &oc.Interface{Name: ygot.String(srcp.Name())}
+	tc.configSrcDUT(srci, &dutSrc)
+	srci.Type = ethernetCsmacd
+	srciPath := d.Interface(srcp.Name())
+	if *deviations.ExplicitPortSpeed {
+		srci.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, srcp)
+	}
+	fptest.LogQuery(t, srcp.String(), srciPath.Config(), srci)
+	gnmi.Replace(t, tc.dut, srciPath.Config(), srci)
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		fptest.AssignToNetworkInstance(t, tc.dut, srcp.Name(), *deviations.DefaultNetworkInstance, 0)
+	}
+
+	for _, port := range tc.dutPorts[1:] {
+		i := &oc.Interface{Name: ygot.String(port.Name())}
+		i.Type = ethernetCsmacd
+
+		if *deviations.InterfaceEnabled {
+			i.Enabled = ygot.Bool(true)
+		}
+
+		tc.configDstMemberDUT(i, port)
+		iPath := d.Interface(port.Name())
+		if *deviations.ExplicitPortSpeed {
+			i.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, port)
+		}
+		fptest.LogQuery(t, port.String(), iPath.Config(), i)
+		gnmi.Replace(t, tc.dut, iPath.Config(), i)
+	}
+}
+
+// configureATE configures source and destination port of ATE and add creates port-channel as well.
+func (tc *testArgs) configureATE(t *testing.T) {
+	t.Helper()
+	if len(tc.atePorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got: %v", tc.atePorts)
+	}
+
+	p0 := tc.atePorts[0]
+	tc.top.Ports().Add().SetName(p0.ID())
+	d0 := tc.top.Devices().Add().SetName(ateSrc.Name)
+	srcEth := d0.Ethernets().Add().SetName(ateSrc.Name + ".Eth").SetMac(ateSrc.MAC)
+	srcEth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(p0.ID())
```

Comment:
SetChoice should be unnecessary here since I believe SetPortName already sets the choice.

Reply:
Connection/Choice doesn't have any defaults and calling SetPortName outside of "Connection" is deprecated. Also Connection.SetPortName without setting the choice will trigger an error.

Reply:
Latest gosnappi (https://github.com/open-traffic-generator/snappi/blob/main/gosnappi/gosnappi.go at https://github.com/open-traffic-generator/snappi/commit/05c4a89a906eb0d9e842f0d9a5dba56114a3e81d) does set the choice on the 'Connection' object automatically (line 60736 for the function).

Reply:
Removed SetChoice